### PR TITLE
fix(#3932)

### PR DESCRIPTION
**Pull request by haverchuck**
*Issue #, if available:*
#3932 

*Description of changes:*
Adds try/catch block to handleAuthResponse.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/auth/src/OAuth/OAuth.ts [hunk 1]
```diff
@@ -183,29 +183,33 @@ export default class OAuth {
...
+          .map(entry => entry.split('='))
+          .reduce((acc, [k, v]) => (acc[k] = v, acc), {}),
+        ...(parse(currentUrl).query || '')
+          .split('&')
+          .map(entry => entry.split('='))
+          .reduce((acc, [k, v]) => (acc[k] = v, acc), {})
+      } as any : {};
+      const { error, error_description } = urlParams;
+  
+      if (error) {
```
Comment (sammartinez) [inline packages/auth/src/OAuth/OAuth.ts]: Is this needed now since we will have the catch for this?

Comment (haverchuck) [reply packages/auth/src/OAuth/OAuth.ts]: @sammartinez -  I believe so?  The 'error' in the if test appears to be a destructured value from the a JSON response, which would not necessarily result in an error in the stack.  It's just a value.  So I would think we still need to manually throw the error.  In any case, I don't think there is harm in leaving it.

File: packages/auth/src/OAuth/OAuth.ts [hunk 2]
```diff
@@ -183,29 +183,33 @@ export default class OAuth {
...
+        throw new Error(error_description);
```
Comment (ericclemmons) [inline packages/auth/src/OAuth/OAuth.ts]: Based on #3932, I couldn't tell if this line is what's causing the development-mode error vs. Line 207/209.

Is there a risk in wrapping _too much_ logic (e.g. `urlParams` creation, throwing with `error_description`, `await this._handleCodeFlow`, `await this._handleImplicitFlow`), and losing visibility into other bugs as a result?

**Comment by navilan** [comment]
@sammartinez @haverchuck - An unfortunate side effect of this PR is that errors are now swallowed. We use Hub.listen to listen on errors and handle them, this change has prevented that.  Can we revert this change please or remove lines 199 - 201?

**Pull request merged**